cg/src/domains/management/subjects/subjects.service.js

```diff
@@ -65,7 +68,22 @@ class SubjectsService {
     };
   }
 
-  async listRectificationRequests(requestedPage = 1) { 
+  async getData(subjectId) {
+    const [data] = await this.db('subjects')
+      .join('subject_keys', 'subjects.id', '=', 'subject_keys.subject_id')
+      .select('personal_data')
+      .select('key')
+      .where({ subject_id: subjectId });
+
+    if (!data) {
+      throw new NotFound('Subject not found');
+    }
+    const decryptedData = decryptFromStorage(data.personal_data, data.key);
+    await recordAccessByController(subjectId);
```

Comment:
This records the access from a subject to it's own data, not the manager's access

Reply:
Why the method name is `recordAccessByController` in this case?

Reply:
To differentiate between controller and processor actions